src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8760,10 +8759,6 @@ private bool IsPossibleStatement(bool acceptAccessibilityMods)
                 case SyntaxKind.IdentifierToken:
                     return IsTrueIdentifier();
 
-                case SyntaxKind.CatchKeyword:
-                case SyntaxKind.FinallyKeyword:
-                    return !_isInTry;
```

Comment:
this was actually *highly* undesirable in the parser.  it effectively made the handling of blocks, and 'catch/finally' tokens context-sensitive.  While unlikely to cause bad issues, it means you could end up in some sort of wonky incremental scenario where something like a block might move inside/outside of a try, and get reused, despite the context not being the same (and thus meaning that non-incremental parsing would produce a different tree).